### PR TITLE
Feature/share walk route

### PR DESCRIPTION
**Pull request by enrichp**

### CODE REVIEW

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/walks/WalkFragment.java [hunk 1]
```diff
@@ -56,9 +64,13 @@
     private WalkPets selectedWalkPets;
     private int[] colors;
     private int nextColor;
+    private static FloatingActionButton flSharePetWalkRouteButton;
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/walks/WalkFragment.java:67]: Codacy Issue found: [Static variable definition in wrong order.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5631707)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 9
           

Clones added
============
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/walks/WalkFragment.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/community/groups/PostsFragment.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5631707&bid=18444749)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/walks/WalkFragment.java [hunk 2]
```diff
@@ -67,13 +79,91 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
...
+    @NonNull
+    private File saveBitmap(Bitmap image) {
+        File file = fileCreation();
+        FileOutputStream fOut;
+        try {
+            fOut = new FileOutputStream(file);
+            image.compress(Bitmap.CompressFormat.JPEG, 100, fOut);
+            fOut.flush();
+            fOut.close();
+        } catch (Exception e) {
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/walks/WalkFragment.java:147]: Codacy Issue found: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5631707)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/walks/WalkFragment.java [hunk 3]
```diff
@@ -67,13 +79,91 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
...
+
+    /**
+     * Creation of the file.
+     * @return The file
+     */
+    @NonNull
+    private File fileCreation() {
+        String file_path = Environment.getExternalStorageDirectory().getAbsolutePath() + "/MyPetCare/ScreenShot";
+        File dir = new File(file_path);
+        if(!dir.exists()) {
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/walks/WalkFragment.java:161]: Codacy Issue found: ['if' is not followed by whitespace.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5631707)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/walks/WalkFragment.java (hunk 2 above)
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/walks/WalkFragment.java:147]: Codacy Issue found: [Catching 'Exception' is not allowed.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5631707)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/walks/WalkFragment.java [hunk 4]
```diff
@@ -67,13 +79,91 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
...
         return binding.getRoot();
     }
 
+    /**
+     * Set the share app button listener.
+     */
+    private void setUpSharePetWalkRouteListener() {
+        flSharePetWalkRouteButton.setOnClickListener(v -> {
+            GoogleMap.SnapshotReadyCallback callback = new GoogleMap.SnapshotReadyCallback() {
+            @Override
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/walks/WalkFragment.java:98]: Codacy Issue found: ['method def modifier' has incorrect indentation level 12, expected level should be one of the following: 16, 20.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5631707)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/walks/WalkFragment.java [hunk 5]
```diff
@@ -67,13 +79,91 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
...
+        return file;
+    }
+
+    /**
+     * Creation of the file.
+     * @return The file
+     */
+    @NonNull
+    private File fileCreation() {
+        String file_path = Environment.getExternalStorageDirectory().getAbsolutePath() + "/MyPetCare/ScreenShot";
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/walks/WalkFragment.java:159]: Codacy Issue found: [Name 'file_path' must match pattern '^[a-z][a-zA-Z0-9]*$'.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5631707)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/walks/WalkFragment.java [hunk 6]
```diff
@@ -67,13 +79,91 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
...
+     * Save a bitmap.
+     * @param image The bitmap of the image
+     * @return The file created
+     */
+    @NonNull
+    private File saveBitmap(Bitmap image) {
+        File file = fileCreation();
+        FileOutputStream fOut;
+        try {
+            fOut = new FileOutputStream(file);
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/walks/WalkFragment.java:143]: Codacy Issue found: [Avoid instantiating FileInputStream, FileOutputStream, FileReader, or FileWriter](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5631707)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/walks/WalkFragment.java [hunk 7]
```diff
@@ -67,13 +79,91 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
...
+
+    /**
+     * Save a bitmap.
+     * @param image The bitmap of the image
+     * @return The file created
+     */
+    @NonNull
+    private File saveBitmap(Bitmap image) {
+        File file = fileCreation();
+        FileOutputStream fOut;
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/walks/WalkFragment.java:141]: Codacy Issue found: [Ensure that resources like this FileOutputStream object are closed after use](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5631707)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/walks/WalkFragment.java [hunk 8]
```diff
@@ -67,13 +79,91 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
...
+        try {
+            fOut = new FileOutputStream(file);
+            image.compress(Bitmap.CompressFormat.JPEG, 100, fOut);
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/walks/WalkFragment.java:144]: Codacy Issue found: ['100' is a magic number.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5631707)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 9
           

Clones added
============
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/walks/WalkFragment.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5631707&bid=18444749)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/walks/WalkFragment.java [hunk 9]
```diff
@@ -87,15 +99,91 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
...
+
+    /**
+     * Creation of the file.
+     * @return The file
+     */
+    @NonNull
+    private File fileCreation() {
+        String file_path = Environment.getExternalStorageDirectory().getAbsolutePath() + "/MyPetCare/ScreenShot";
+        File dir = new File(file_path);
+        if(!dir.exists()) {
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/walks/WalkFragment.java:181]: Codacy Issue found: ['if' is not followed by whitespace.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5631707)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/walks/WalkFragment.java [hunk 10]
```diff
@@ -87,15 +99,91 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
...
+    @NonNull
+    private File saveBitmap(Bitmap image) {
+        File file = fileCreation();
+        FileOutputStream fOut;
+        try {
+            fOut = new FileOutputStream(file);
+            image.compress(Bitmap.CompressFormat.JPEG, 100, fOut);
+            fOut.flush();
+            fOut.close();
+        } catch (Exception e) {
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/walks/WalkFragment.java:167]: Codacy Issue found: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5631707)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/walks/WalkFragment.java [hunk 11]
```diff
@@ -60,10 +68,14 @@
     private WalkPets selectedWalkPets;
     private int[] colors;
     private int nextColor;
+    private static FloatingActionButton flSharePetWalkRouteButton;
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/walks/WalkFragment.java:71]: Codacy Issue found: [Static variable definition in wrong order.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5631707)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/walks/WalkFragment.java [hunk 12]
```diff
@@ -87,15 +99,91 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
...
+     * Save a bitmap.
+     * @param image The bitmap of the image
+     * @return The file created
+     */
+    @NonNull
+    private File saveBitmap(Bitmap image) {
+        File file = fileCreation();
+        FileOutputStream fOut;
+        try {
+            fOut = new FileOutputStream(file);
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/walks/WalkFragment.java:163]: Codacy Issue found: [Avoid instantiating FileInputStream, FileOutputStream, FileReader, or FileWriter](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5631707)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/walks/WalkFragment.java [hunk 13]
```diff
@@ -87,15 +99,91 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
...
         return binding.getRoot();
     }
 
+    /**
+     * Set the share app button listener.
+     */
+    private void setUpSharePetWalkRouteListener() {
+        flSharePetWalkRouteButton.setOnClickListener(v -> {
+            GoogleMap.SnapshotReadyCallback callback = new GoogleMap.SnapshotReadyCallback() {
+            @Override
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/walks/WalkFragment.java:118]: Codacy Issue found: ['method def modifier' has incorrect indentation level 12, expected level should be one of the following: 16, 20.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5631707)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/walks/WalkFragment.java (hunk 10 above)
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/walks/WalkFragment.java:167]: Codacy Issue found: [Catching 'Exception' is not allowed.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5631707)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/walks/WalkFragment.java [hunk 14]
```diff
@@ -87,15 +99,91 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
...
+
+    /**
+     * Save a bitmap.
+     * @param image The bitmap of the image
+     * @return The file created
+     */
+    @NonNull
+    private File saveBitmap(Bitmap image) {
+        File file = fileCreation();
+        FileOutputStream fOut;
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/walks/WalkFragment.java:161]: Codacy Issue found: [Ensure that resources like this FileOutputStream object are closed after use](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5631707)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/walks/WalkFragment.java [hunk 15]
```diff
@@ -87,15 +99,91 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
...
+        return file;
+    }
+
+    /**
+     * Creation of the file.
+     * @return The file
+     */
+    @NonNull
+    private File fileCreation() {
+        String file_path = Environment.getExternalStorageDirectory().getAbsolutePath() + "/MyPetCare/ScreenShot";
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/walks/WalkFragment.java:179]: Codacy Issue found: [Name 'file_path' must match pattern '^[a-z][a-zA-Z0-9]*$'.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5631707)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/walks/WalkFragment.java [hunk 16]
```diff
@@ -87,15 +99,91 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
...
+     * @param image The bitmap of the image
+     * @return The file created
+     */
+    @NonNull
+    private File saveBitmap(Bitmap image) {
+        File file = fileCreation();
+        FileOutputStream fOut;
+        try {
+            fOut = new FileOutputStream(file);
+            image.compress(Bitmap.CompressFormat.JPEG, 100, fOut);
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/walks/WalkFragment.java:164]: Codacy Issue found: ['100' is a magic number.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5631707)

**Review by AlbertPG — DISMISSED** [review]
Okey, check for Codacy and it will be perfect

**Review by AlbertPG — APPROVED** [review]
Okey

**Review by xavier-campos-diaz — APPROVED** [review]

**Pull request merged**